pkg/network/master/vnids.go

```diff
@@ -304,18 +301,31 @@ func (master *OsdnMaster) handleDeleteNamespace(obj interface{}) {
 }
 
 func (master *OsdnMaster) watchNetNamespaces() {
-	common.RunEventQueue(master.networkClient.Network().RESTClient(), common.NetNamespaces, func(delta cache.Delta) error {
-		netns := delta.Object.(*networkapi.NetNamespace)
-		name := netns.ObjectMeta.Name
-
-		log.V(5).Infof("Watch %s event for NetNamespace %q", delta.Type, name)
-		switch delta.Type {
-		case cache.Sync, cache.Added, cache.Updated:
-			err := master.vnids.updateVNID(master.networkClient, netns)
-			if err != nil {
-				return fmt.Errorf("error updating netid: %v", err)
-			}
-		}
-		return nil
-	})
+	common.RegisterSharedInformer(master.informers, master.handleAddOrUpdateNetNamespace, master.handleDeleteNetNamespace, common.NetNamespaces)
+}
+
+func (master *OsdnMaster) handleAddOrUpdateNetNamespace(obj, _ interface{}, eventType watch.EventType) {
+	origNetns := obj.(*networkapi.NetNamespace)
+	log.V(5).Infof("Watch %s event for NetNamespace %q", eventType, origNetns.Name)
+
+	// Informer cache should not be mutated, so get a copy of the object
+	objCopy, err := kapi.Scheme.DeepCopy(origNetns)
+	if err != nil {
+		log.Errorf("Failed to deep copy NetNamespace %v", origNetns)
+		return
+	}
+	netns, ok := objCopy.(*networkapi.NetNamespace)
+	if !ok {
```

Comment:
Just assume that it's the right type. (We don't double-check the typecast at the start of the function, so there's no reason to double-check this one either.)